homeassistant/components/weather/__init__.py

```diff
@@ -207,13 +366,26 @@ def precision(self) -> float:
     def state_attributes(self):
         """Return the state attributes, converted from native units to user-configured units."""
         data = {}
-        if self.temperature is not None:
-            data[ATTR_WEATHER_TEMPERATURE] = show_temp(
-                self.hass,
-                self.temperature,
-                self.temperature_unit,
-                self.precision,
-            )
+
+        precision_str = str(self.precision)
+        precision = (
+            len(precision_str) - precision_str.index(".") - 1
+            if "." in precision_str
+            else 0
+        )
+
+        if (temperature := self.temperature) is not None:
+            with suppress(ValueError):
+                float(temperature)
```

Comment:
This doesn't do anything, you maybe want to do:
```suggestion
                temperature_f = float(temperature)
```
And then use the `temperature_f ` below in the conversion